jib-build-plan/src/main/java/com/google/cloud/tools/jib/api/buildplan/FilePermissionsProvider.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.api.buildplan;
+
+import java.nio.file.Path;
+
+/** Interface for providing rules to determine file permissions on a container. */
+@FunctionalInterface
+public interface FilePermissionsProvider {
+
+  /**
+   * Returns the file permissions that should be set for a path, given the source path and
+   * destination path on a container.
+   *
+   * @param sourcePath the source file.
+   * @param destinationPath the destination path in the container file system corresponding to
+   *     sourcePath.
+   * @return the permissions to be set for the file.
```

Comment:
style super-nit

Although not really enforced, I generally follow these guidelines.

> When writing the comments themselves, in general, start with a phrase and follow it with sentences if they are needed.
> 
> - When writing a phrase, do not capitalize and do not end with a period:
> ```
> @param x the x-coordinate, measured in pixels
> ```
> 
> - When writing a phrase followed by a sentence, do not capitalize the phrase, but end it with a period to distinguish it from the start of the next sentence:
> ```
> @param x the x-coordinate. Measured in pixels.
> ```
> 
> - If you prefer starting with a sentence, capitalize it and end it with a period:
> ```
> @param x Specifies the x-coordinate, measured in pixels.
> ```
> 
> - When writing multiple sentences, follow normal sentence rules:
> ```
> @param x Specifies the x-coordinate. Measured in pixels.
> ```
>

Reply:
Ah this does make it easier to understand what the parameters mean.